### PR TITLE
Improved Unit Test for ByteString

### PR DESCRIPTION
**Pull request by gleichdick**
### Description

Improved Unit test for ByteString class of py_bindings_tools. Superseeds #1983.

P.S. please don't mind the branch name, this PR has nothing to do with the GILReleaser.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning_interface/test/CMakeLists.txt [hunk 1]
```diff
@@ -22,4 +22,15 @@ if (CATKIN_ENABLE_TESTING)
   add_rostest(python_move_group_ns.test)
   add_rostest(robot_state_update.test)
   add_rostest(cleanup.test)
+
+  add_library(moveit_planning_interface_test_serialize_msg_cpp_helper_python serialize_msg_python_cpp_helpers.cpp)
```
Comment (rhaschke) [inline moveit_ros/planning_interface/test/CMakeLists.txt:26]: What about introducing a variable, e.g. `HELPER_LIB`, for this monster of library name?

File: moveit_ros/planning_interface/test/serialize_msg.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    def test_rejectUnicodeTuple(self):
+        with self.assertRaisesRegexp(RuntimeError, "Underlying python object is not a Bytes/String instance"):
+            self.helper.compareVectorTuple((u'kdasd',))
+
+
+if __name__ == '__main__':
+    import rostest
+    PKGNAME = 'moveit_ros_planning_interface'
+    NODENAME = 'moveit_test_message_serialisation'
+    rostest.rosrun(PKGNAME, NODENAME, PythonMsgSerializeTest)
```
Comment (rhaschke) [inline moveit_ros/planning_interface/test/serialize_msg.py:113]: Doesn't it suffice to call all unittests? Do you really require ROS functionality here?
```suggestion
    unittest.main()
```

**Review by rhaschke — APPROVED** [review]
Great improvement! Thanks for this effort. I have a few more nitpicks though.
Otherwise, I approve.

File: moveit_ros/planning_interface/test/serialize_msg_python_cpp_helpers.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    s.deserialize(v);
+    return v.x == 1.0 && v.y == -2.0 && v.z == 0.25;
+  }
+
+  static void setup()
+  {
+    bp::class_<ByteStringTestHelper> cls("ByteStringTestHelper");
+    cls.def("compareEmbeddedZeros", &ByteStringTestHelper::compareEmbeddedZeros);
+    cls.def("compareTuple", &ByteStringTestHelper::compareTuple);
+    cls.def("compareVectorTuple", &ByteStringTestHelper::compareVectorTuple);
```
Comment (rhaschke) [inline moveit_ros/planning_interface/test/serialize_msg_python_cpp_helpers.cpp:123]: This is defined twice: first occurence here.

File: moveit_ros/planning_interface/test/serialize_msg_python_cpp_helpers.cpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    cls.def("getBytesPChar", &ByteStringTestHelper::getBytesPChar);
+    cls.def("getBytesStdString", &ByteStringTestHelper::getBytesStdString);
+    cls.def("getDefaultBytes", &ByteStringTestHelper::getDefaultBytes);
+    cls.def("getTuple", &ByteStringTestHelper::getTuple);
+    cls.def("getVector", &ByteStringTestHelper::getVector);
+    cls.def("compareVector", &ByteStringTestHelper::compareVector);
+    cls.def("compareVectorTuple", &ByteStringTestHelper::compareVectorTuple);
```
Comment (rhaschke) [inline moveit_ros/planning_interface/test/serialize_msg_python_cpp_helpers.cpp:130]: 2nd occurence here.

File: moveit_ros/planning_interface/test/serialize_msg_python_cpp_helpers.cpp [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    geometry_msgs::Vector3 v;
+    s.deserialize(v);
+    return v.x == 1.0 && v.y == -2.0 && v.z == 0.25;
+  }
+  bool compareVectorTuple(const bp::tuple& t)
+  {
+    const ByteString s(t[0]);
+    geometry_msgs::Vector3 v;
+    s.deserialize(v);
+    return v.x == 1.0 && v.y == -2.0 && v.z == 0.25;
```
Comment (rhaschke) [inline moveit_ros/planning_interface/test/serialize_msg_python_cpp_helpers.cpp:115]:
```suggestion
    return compareVector(s);
```

**Review by rhaschke — CHANGES_REQUESTED** [review]
Just noted some more glitches. Sorry for being that picky.

**Comment by gleichdick** [comment]
No no, that's why we do reviews :smile:

**Review by rhaschke — APPROVED** [review]
Thanks for fixing.

**Pull request merged**